plugin-viewport/src/main/kotlin/com/mapbox/maps/plugin/viewport/ViewportPluginImpl.kt

```diff
@@ -0,0 +1,389 @@
+   *
+   * The target camera position is obtained with [ViewportDataSource.getViewportData].
+   *
+   * @param stateTransitionOptionsBlock options that impact the transition animation from
+   * the current state to the requested state.
+   * Defaults to [ViewportCameraTransitionOptions.maxDurationMs] equal to 3500 millis.
+   * @param frameTransitionOptionsBlock options that impact the transition animations between
+   * viewport frames in the selected state.
+   * This refers to camera transition on each [ViewportDataSource] update when [Following] is engaged.
+   * Defaults to [ViewportCameraTransitionOptions.maxDurationMs] equal to 1000 millis.
+   * @param transitionEndListener invoked when transition ends.
+   */
+  @JvmOverloads
+  override fun requestNavigationCameraToFollowing(
+    stateTransitionOptionsBlock: ((ViewportCameraTransitionOptions.Builder).() -> Unit),
+    frameTransitionOptionsBlock: ((ViewportCameraTransitionOptions.Builder).() -> Unit),
+    transitionEndListener: TransitionEndListener?,
+  ) {
+    requestNavigationCameraToFollowing(
+      ViewportCameraTransitionOptions.Builder().apply(stateTransitionOptionsBlock).build(),
+      ViewportCameraTransitionOptions.Builder().apply(frameTransitionOptionsBlock).build(),
+      transitionEndListener,
+    )
+  }
+
+  /**
+   * Executes a transition to [Following] state. When started, goes to [TransitionToFollowing]
+   * and to the final [Following] when ended. If transition is canceled, state goes to [Idle].
+   *
+   * The target camera position is obtained with [ViewportDataSource.getViewportData].
+   *
+   * @param stateTransitionOptions options that impact the transition animation from the current state to the requested state.
+   * Defaults to [ViewportCameraTransitionOptions.maxDurationMs] equal to 3500 millis.
+   * @param frameTransitionOptions options that impact the transition animations between viewport frames in the selected state.
+   * This refers to camera transition on each [ViewportDataSource] update when [Following] is engaged.
+   * Defaults to [ViewportCameraTransitionOptions.maxDurationMs] equal to 1000 millis.
+   * @param transitionEndListener invoked when transition ends.
+   */
+  @JvmOverloads
+  override fun requestNavigationCameraToFollowing(
+    stateTransitionOptions: ViewportCameraTransitionOptions,
+    frameTransitionOptions: ViewportCameraTransitionOptions,
+    transitionEndListener: TransitionEndListener?,
+  ) {
+    when (state) {
+      TransitionToFollowing -> transitionEndListener?.let { listener ->
+        transitionEndListeners.add(listener)
+      }
+      Following -> transitionEndListener?.onTransitionEnd(isCanceled = false)
+      Idle, TransitionToOverview, Overview -> {
+        val data = viewportDataSource.getViewportData()
+        startAnimation(
+          stateTransition.transitionToFollowing(data.cameraForFollowing, stateTransitionOptions)
+            .apply {
+              addListener( // todo
```

Comment:
add more information about todo

Reply:
applies to methods below

Reply:
@axti could you give more context on what's left?